src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,10 +34,33 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+
+        //Split address into Block, Street, Unit and Postal code
+        String[] addressPortions = splitAddress(trimmedAddress);
+
+        if (!isValidAddress(trimmedAddress) || (addressPortions.length != NUMBER_OF_PORTIONS)) {
```

Comment:
Quite okay but sometimes address may not have a valid block etc, is it wise to strictly enforce all fields must be entered?